dataframe-jdbc/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/readJdbc.kt

```diff
@@ -775,6 +781,9 @@ private fun manageColumnNameDuplication(columnNameCounter: MutableMap<String, In
     return name
 }
 
+// Utility function to cast arrays based on the type of elements
+public fun <T : Any> castArray(array: Array<*>, elementType: KClass<T>): List<T> = array.map { elementType.cast(it) }
```

Comment:
does this need to be public? (also probably change the comment to kdoc so it shows up in the IDE)